pkg/controller/installation/types.go

```diff
@@ -122,15 +122,15 @@ func (t *Type) GetStages() []Stage {
 	return t.Stages
 }
 
-func InstallationTypeFactory(installationType string, products []string) (error, *Type) {
+func TypeFactory(installationType string, products []string) (*Type, error) {
```

Comment:
For the name here, it's saying that it's not great to have `installation.InstallationTypeFactory`, and `installation.TypeFactory` would be better, right?

In this case, I can't see any usage of it outside this package, so we only ever say `TypeFactory`. If I'm correct, then I think it would be better to unexport this, and keep the name the same. WDYT?

In essence:
```suggestion
func installationTypeFactory(installationType string, products []string) (*Type, error) {
```

Reply:
Think this was resolved in another PR, the recent rebase already has this implemented it seems